rg_instructor_analytics/views/Problem.py

```diff
@@ -0,0 +1,250 @@
+                if child.location.category == 'problem':
+                    problem_id = child.location.to_deprecated_string()
+                    if problem_id in academic_performance:
+                        current_performance = academic_performance[problem_id]
+                        stat['correct_answer'][-1] += current_performance['grade_avg']
+                        stat['attempts'][-1] += current_performance['attempts_avg']
+                        problems_in_hw += 1
+
+                    stat['problems'][-1].append(problem_id)
+
+            if problems_in_hw > 0:
+                stat['correct_answer'][-1] /= problems_in_hw
+                stat['attempts'][-1] /= problems_in_hw
+
+        return stat
+
+    def process(self, request, **kwargs):
+        """
+        Process post request.
+        """
+        return JsonResponse(data=self.get_homework_stat(kwargs['course_key']))
+
+
+class ProblemsStatisticView(AccessMixin, View):
+    """
+    Api for getting statistic for each problem in unit.
+    """
+
+    def process(self, request, **kwargs):
+        """
+        Process post request.
+        """
+        course_key = kwargs['course_key']
+        problems = [course_key.make_usage_key_from_deprecated_string(p) for p in request.POST.getlist('problems')]
+        stats = (
+            StudentModule.objects
+            .filter(module_state_key__in=problems)
+            .values('module_state_key')
+            .annotate(correct=Sum('grade'))
+            .annotate(incorrect=Sum('grade') - Sum('max_grade'))
+            .values('module_state_key', 'correct', 'incorrect')
+        )
+        incorrect, correct = tuple(map(list, zip(*[(int(s['incorrect'] or 0), int(s['correct'] or 0)) for s in stats])))
+        return JsonResponse(data={'incorrect': incorrect, 'correct': correct})
+
+
+class ProblemDetailView(AccessMixin, View):
+    """
+    Api for getting problem detail.
+    """
+
+    def process(self, request, **kwargs):
+        """
+        Process post request.
+        """
+        return xblock_view(request, kwargs['course_id'], request.POST['problem'], 'student_view')
+
+
+class ProblemQuestionParser:
+    """
+    Base class for provide statistic for question.
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, problemID, questionID, answer_map):
+        """
+        Constructor.
+
+        :param problemID: object (not string), that represent problem ID
+        :param questionID: string that represent question
+        :param answer_map: map for link data from data base to real data
+        """
+        self.answer_map = answer_map
+        self.problemID = problemID
+        self.questionID = questionID
+
+    @staticmethod
+    def init_statistic_object():
+        """
+        Provide init state for statistic.
+        """
+        return {}
+
+    @abstractmethod
+    def process_statistic_item(self, state, item):
```

Comment:
This class doesn't look like an `interface` but like the base class. I would use NotImplementedError instead of the abstractmethod decorator.

Reply:
Why? This class already abstract and something like NotImplementedError will be raised, when implementation missed.